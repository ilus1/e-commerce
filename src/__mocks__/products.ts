export const mockProducts = [
  { id: '1', name: 'Produto 1', price: 10, photo: 'url1', description: 'Descrição 1' },
];

export const mockCartItems = [
  { id: '1', name: 'Produto 1', price: 10, photo: 'url1', description: 'Descrição 1', count: 1 },
];
