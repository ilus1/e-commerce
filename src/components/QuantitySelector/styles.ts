import styled from "styled-components"

import colors from "../../constants/colors"

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  font-size: 8px;

  width: 50px;
  min-height: 19px;
  border-radius: 4px;
  border: 0.3px solid ${colors.quantityBorderColor};
  background-color: ${colors.background};
`

export const Quantity = styled.p`
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
`

export const Button = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;

  width: 50%;
  height: 100%;
  background-color: ${colors.background};
  color: ${colors.primary};

  border: none;
  border-radius: 5px;
  cursor: pointer;

  &:hover {
    background-color: ${colors.footer};
  }

  &:active {
    background-color: ${colors.primary};
    color: ${colors.background};
  }
`
