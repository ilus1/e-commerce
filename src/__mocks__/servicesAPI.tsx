import { rest } from "msw";
import { setupServer } from "msw/node";

import { mockProducts } from "./products";

const server = setupServer(
  rest.get('*', (_, res, ctx) => res(ctx.status(200), ctx.json(mockProducts)))
);

export default server;
