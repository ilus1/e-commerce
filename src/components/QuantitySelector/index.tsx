import { Container, Button, Quantity } from './styles';

interface CartItem {
  id: string
  name: string
  price: number
  photo: string
  description: string
  count: number
}

interface QuantitySelectorProps {
  item: CartItem
  onAdd: (item: CartItem) => void
  onRemove: (item: CartItem) => void
}

function QuantitySelector({ item, onAdd, onRemove }: QuantitySelectorProps) {
  const { count } = item;

  return (
    <Container>
      <Button onClick={() => onRemove(item)}>-</Button>
      <Quantity>{count}</Quantity>
      <Button onClick={() => onAdd(item)}>+</Button>
    </Container>
  );
}

export default QuantitySelector;
