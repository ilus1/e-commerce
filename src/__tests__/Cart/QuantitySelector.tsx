import renderer from 'react-test-renderer'
import { render, screen, fireEvent } from '@testing-library/react';
import QuantitySelector from '../../components/QuantitySelector';

import { mockCartItems } from '../../__mocks__/products';

describe.only('Componente de alterar quantidade de um item no carrinho', () => {

  it('Testa carregamento correto', () => {
    const tree = renderer
    .create(
      <QuantitySelector
        item={mockCartItems[0]}
        onAdd={() => {}}
        onRemove={() => {}}
      />
    );

    expect(tree).toMatchSnapshot();
  });

  it('Testa botao de aumentar quantidade do produto', () => {
    const mockOnAdd = jest.fn();
    render(
      <QuantitySelector
        item={mockCartItems[0]}
        onAdd={mockOnAdd}
        onRemove={() => {}}
      />
    );

    fireEvent.click(screen.getByText('+'));
    expect(mockOnAdd).toHaveBeenCalledWith(mockCartItems[0]);
  });

  it('Testa botao de diminuir quantidade do produto', () => {
    const mockOnRemove = jest.fn();
    render(
      <QuantitySelector
        item={mockCartItems[0]}
        onAdd={() => {}}
        onRemove={mockOnRemove}
      />
    );

    fireEvent.click(screen.getByText('-'));
    expect(mockOnRemove).toHaveBeenCalledWith(mockCartItems[0]);
  });
});
