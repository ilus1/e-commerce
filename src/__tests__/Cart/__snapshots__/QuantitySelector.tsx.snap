// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de alterar quantidade de um item no carrinho Testa carregamento correto 1`] = `
<div
  className="sc-bdfCDU fhrPBJ"
>
  <button
    className="sc-dlfmHC jhxSjB"
    onClick={[Function]}
  >
    -
  </button>
  <p
    className="sc-gsTDqH iORjDB"
  >
    1
  </p>
  <button
    className="sc-dlfmHC jhxSjB"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
